### PR TITLE
Document and Script Releases

### PR DESCRIPTION
**Pull request by ghickman**
Document how to make a release and wrap the PyPI steps up in a makefile to make `make release` possible.

I've added `make docs` as a shortcut for running the documentation engine too.

### CODE REVIEW

File: doc/docs/guides/releasing.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Making a Release
+1. Tag the HEAD of the release branch and push to github: `git tag <version> && git push --tags`
+
+1. Check version is correct in `_version.py`.
+
+1. Make sure you have `twine` installed and then: `make release`
```
Comment (fredkingham) [inline doc/docs/guides/releasing.md]: @ghickman  why not add twine to the requirements?

Comment (ghickman) [reply doc/docs/guides/releasing.md]: I considered that but it's only necessary for people who are uploading releases to PyPI so seemed superfluous for typical development.  Your call though!

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	@echo "    make docs    -- setup and serve docs locally"
+	@echo "    make release -- build and release to PyPI"
+
+docs:
+	pip install -r doc/requirements.txt
+	mkdocs serve
+
+release:
+	rm -rf dist/*
+	python setup.py register bdist_wheel sdist
```
Comment (davidmiller) [inline Makefile]: Why register here ? I thought you only need to do that once ?

File: doc/docs/guides/releasing.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Making a Release
+1. Tag the HEAD of the release branch and push to github: `git tag <version> && git push --tags`
+
+1. Check version is correct in `_version.py`.
```
Comment (davidmiller) [inline doc/docs/guides/releasing.md]: Are you deliberately writing a numbered list with all the numbers as 1. 1. 1. ?

**Comment by davidmiller** [comment]
Mostly this looks great - couple comments and I've added some further steps in #1450

File: doc/docs/guides/releasing.md (hunk 1 above)
Comment (davidmiller) [reply doc/docs/guides/releasing.md]: I think this is fine tbh - let's see if it actually trips anyone up :)

File: doc/docs/guides/releasing.md (hunk 3 above)
Comment (pacharanero) [reply doc/docs/guides/releasing.md]: IIUC this is correct Markdown, any markdown preprocessor will auto-renumber a 1. 1. 1. 1. list as 1. 2. 3. 4, which makes it easier if you needed to re-order the items in the list ever.

I guess the downside is that anyone viewing the unpreprocessed source will not see a correctly numbered list.

Comment (ghickman) [reply doc/docs/guides/releasing.md]: @pacharanero is correct here, it's how one does a markdown list for auto-numbering.  It's definitely a bit odd at first view but if you ever have to insert numbers into a list it's much easier!

File: Makefile (hunk 2 above)
Comment (ghickman) [reply Makefile]: Register sets metadata on the PyPI so if you ever change something like Trove classifiers or author email in your setup.py it wouldn't be updated in the PyPI until you run `register` again.

Doing it on every release is probably overkill but it removes you ever having to think about it and is harmless when nothing has changed.

Comment (davidmiller) [reply Makefile]: I did not know that upload wouldn't update metadata !

**Pull request merged**